Input: Refactor +use handling

saibotk opened the pull request:
This will get a proper description & commits later.

Fixes #1816

wgetJane commented:
skimmed through the changes, are you sure these won't reintroduce any of the bugs fixed by #1792?

saibotk commented:
Pretty sure, as all the changes that i removed were mainly introduced to workaround the broken use system.
But will check doors etc. on the weekend

saibotk commented:
Okay @TimGoll , i finished this one up and fixed any remaining bugs i could find when testing this locally.

Also added a description and changelog!

The change is actually rather simple now, but understanding the code parts can be hard, if you have any questions ask me and i'll happily help out!
